pandas/core/arrays/interval.py

```diff
@@ -0,0 +1,814 @@
+    of all intervals:
+
+    >>> pd.Index([pd.Interval(0, 1), pd.Interval(1, 2)])
+    %(klass)s([(0, 1], (1, 2]]
+                  closed='right', dtype='interval[int64]')
+
+    See Also
+    --------
+    interval_range : Function to create a fixed frequency IntervalIndex
+    %(klass)s.from_arrays : Construct an %(klass)s from a left and
+                                right array
+    %(klass)s.from_breaks : Construct an %(klass)s from an array of
+                                splits
+    %(klass)s.from_tuples : Construct an %(klass)s from a
+                                list/array of tuples
+    """
+
+    _interval_shared_docs['from_tuples'] = """
+    Construct an %(klass)s from a list/array of tuples
+
+    Parameters
+    ----------
+    data : array-like (1-dimensional)
+        Array of tuples
+    closed : {'left', 'right', 'both', 'neither'}, default 'right'
+        Whether the intervals are closed on the left-side, right-side, both
+        or neither.
+    copy : boolean, default False
+        by-default copy the data, this is compat only and ignored
+    dtype : dtype or None, default None
+        If None, dtype will be inferred
+
+        ..versionadded:: 0.23.0
+
+
+    Examples
+    --------
+    >>>  pd.%(klass)s.from_tuples([(0, 1), (1, 2)])
+    %(klass)s([(0, 1], (1, 2]],
+                closed='right', dtype='interval[int64]')
+
+    See Also
+    --------
+    interval_range : Function to create a fixed frequency IntervalIndex
+    %(klass)s.from_arrays : Construct an %(klass)s from a left and
+                                right array
+    %(klass)s.from_breaks : Construct an %(klass)s from an array of
+                                splits
+    """
+
+    @classmethod
+    @Appender(_interval_shared_docs['from_tuples'] % _shared_docs_kwargs)
+    def from_tuples(cls, data, closed='right', copy=False, dtype=None):
+        if len(data):
+            left, right = [], []
+        else:
+            left = right = data
+
+        for d in data:
+            if isna(d):
+                lhs = rhs = np.nan
+            else:
+                name = cls.__name__
+                try:
+                    # need list of length 2 tuples, e.g. [(0, 1), (1, 2), ...]
+                    lhs, rhs = d
+                except ValueError:
+                    msg = ('{name}.from_tuples requires tuples of '
+                           'length 2, got {tpl}').format(name=name, tpl=d)
+                    raise ValueError(msg)
+                except TypeError:
+                    msg = ('{name}.from_tuples received an invalid '
+                           'item, {tpl}').format(name=name, tpl=d)
+                    raise TypeError(msg)
+                lhs, rhs = d
+            left.append(lhs)
+            right.append(rhs)
+
+        return cls.from_arrays(left, right, closed, copy=False,
+                               dtype=dtype)
+
+    def _validate(self):
+        """Verify that the IntervalIndex is valid.
```

Comment:
see my comment above about setting closed. I *think* you *always* validate before calling _simple_new (though maybe not in the main constructor)?